components/style/invalidation/element/element_wrapper.rs

```diff
@@ -182,9 +182,9 @@ impl<'a, E> Element for ElementWrapper<'a, E>
             // FIXME(bz): How can I set this up so once Servo adds :dir()
             // support we don't forget to update this code?
             #[cfg(feature = "gecko")]
-            NonTSPseudoClass::Dir(ref s) => {
+            NonTSPseudoClass::Dir(ref dir) => {
                 use invalidation::element::invalidation_map::dir_selector_to_state;
-                let selector_flag = dir_selector_to_state(s);
```

Comment:
Similarly, we can simplify `dir_selector_to_state` to take `&Direction`, instead of comparing the string.

Reply:
That should make the `Box<[u16]>` impl unnecessary.